### ISSUE TITLE
IINA incorrectly reports "No subtitles found" using Open Subtitles

### ISSUE THREAD

**Issue by low-batt** [iina/iina#3907]
**System and IINA version:**

- macOS 12.4
- IINA 1.3.0

**Expected behavior:**
If subtitles can be found by hand using Open Subtitle's web site then IINA will be able to find the subtitles.

**Actual behavior:**
For a video file named `Better.Call.Saul.S04E03.1080p.BluRay.x265-RARBG.mp4` using Open Subtitles IINA reports `No subtitles found`. Searching [Open Subtitles](https://www.opensubtitles.org/en/en) by hand using the text `Better.Call.Saul.S04E03.1080p.BluRay.x265-RARBG`, Open Subtitles adjusts the search string to be `better call saul s04e03` and finds the correct subtitle file.

<br>

**Steps to reproduce:**
- Make a copy of any video long enough to have subtitles
- Rename the copy to be `Better.Call.Saul.S04E03.1080p.BluRay.x265-RARBG.mp4`
- Open the video file in IINA
- Click on `Find Online Subtitles...` under the `Subtitles` menu
- The OSD message `No subtitles found` is displayed

- [x] MPV does not have this problem.

This is an IINA issue.

**How often does this happen?**
Everytime.

**Comment by low-batt** [comment]
### Analysis

The problematic code is the following line from `OpenSubSubtitle.requestIMDB`:
```swift
let IMDB = (bestGuess?["IDMovieIMDB"] as? String) ?? ""
```

That line is expecting Open Subtitles to return a result of this form:
```xml
<name>BestGuess</name>
<value>
  <struct>
    <member>
      <name>IDMovieIMDB</name>
      <value>
        <string>0080339</string>
      </value>
    </member>
```

In the case at hand Open Subtitles is returning this:
```xml
<name>BestGuess</name>
<value>
  <struct>
    <member>
      <name>IDMovieIMDB</name>
      <value>
        <int>3032476</int>
      </value>
    </member>
```

So the cast to `String` in the code shown above quietly fails and the rest of the code assumes Open Subtitles did not find any subtitles.

### Fixing

There are issues with merge conflicts in this area with the plugin system, so I will be holding off creating a pull request until the plugin system has been merged.

The fix not hard. Maybe something like:
```swift
guard let bestGuess = try self.findPath(["data", filename, "BestGuess"], in: response) as? [String: Any], let imdbEntry = bestGuess["IDMovieIMDB"] else {
  resolver.fulfill("")
  return
}
if let IMDB = imdbEntry as? Int {
  resolver.fulfill(String(IMDB))
  return
}
let IMDB = imdbEntry as? String ?? ""
resolver.fulfill(IMDB)
```

**Comment by low-batt** [comment]
Added label to indicate fix has been merged into develop. Re-opening since this is still a problem in IINA 1.3.0, the latest release. This way any users planning on reporting this problem will hopefully find this open issue and not enter a duplicate.

**Comment by low-batt** [comment]
Fixed in IINA 1.3.1.
